src/host/outputStream.cpp

```diff
@@ -202,7 +202,14 @@ bool ConhostInternalGetSet::SetInputMode(const TerminalInput::Mode mode, const b
     // us that SSH 7.7 _also_ requests mouse input and that can have a user interface
     // impact on the actual connected terminal. We can't remove this check,
     // because SSH <=7.7 is out in the wild on all versions of Windows <=2004.
-    return !(IsConsolePty() && IsVtInputEnabled());
+    const bool inConpty = IsConsolePty();
+    const bool shouldPassthrough = inConpty && IsVtInputEnabled();
+
+    // GH#12799 - If the app requested that we disable focus events, DON'T pass
+    // that through. ConPTY would _always_ like to know about focus events.
+    const bool disabledFocusEvents = inConpty && (!enabled && mode == TerminalInput::Mode::FocusEvent);

```

Comment:
FYI purely from a boolean logic, this extra `inConpty` check is unneccessary.
```
!(IsConsolePty && IsVtInputEnabled) || (IsConsolePty && (!enabled && mode == FocusEvent))
```
is equivalent to
```
!IsConsolePty || !IsVtInputEnabled || (!enabled && mode == FocusEvent))
```
Such a expression IMO is also a bit more readable as it's like a "filter" left to right. Due to the early return via `!IsConsolePty` once you're at the `!enabled` part `IsConsolePty` can only be `true` anymore.

Reply:
> I'm not sure why these changes haven't been recognised as a merge conflict

I bet it's because of the way I've stacked like 3 PRs up here. I bet only the lowest layer got caught for the conflict. Resolved most of these now.

Reply:
also added gratuitous comments, because boolean logic is hard some times